operator/pkg/ciliumendpointslice/manager.go

```diff
@@ -118,9 +150,58 @@ func (c *cesManager) RemoveCEPMapping(cep *cilium_v2.CoreCiliumEndpoint, ns stri
 		if c.mapping.countCEPsInCES(cesName) == 0 {
 			c.mapping.deleteCES(cesName)
 		}
-		return NewCESKey(cesName.string(), ns)
+		return []CESKey{NewCESKey(cesName.string(), pod.Namespace)}
+	}
+	return []CESKey{CESKey(resource.Key{})}
+}
+
+func (c *cesManager) UpdateNodeMapping(node *cilium_v2.CiliumNode) []CESKey {
+	newKey := NewEncryptionKey(node.Spec.Encryption.Key)
+	name := NewNodeName(node.Name)
+	if c.mapping.hasNode(name) {
+		// If node encryption key is changed, update affected CES.
+		oldKey := c.mapping.getNodeEncryptionKey(name)
+		if oldKey != newKey {
+			return c.mapping.insertNode(name, newKey)
+		}
+	} else {
+		return c.mapping.insertNode(name, newKey)
+	}
+	return nil
+}
+
+func (c *cesManager) RemoveNodeMapping(node *cilium_v2.CiliumNode) []CESKey {
+	if c.mapping.hasNode(NewNodeName(node.Name)) {
```

Comment:
Nit. The `hasNode` check could be dropped, as `deleteNode` is idempotent. The pre-check could be also potentially racy given that both `hasNode` and `deleteNode` acquire the lock independently, although that is guaranteed not to happen because only one node event is processed at any time.